feflow/protocols/nonequilibrium_cycling.py

```diff
@@ -198,26 +225,26 @@ def _execute(self, ctx, *, state_a, state_b, mapping, settings, **inputs):
             ):
                 common_small_mols[comp] = comp.to_openff()
 
-        # Assign charges to ALL small mols, if unassigned -- more info: Openfe issue #576
-        for off_mol in chain(all_alchemical_mols.values(), common_small_mols.values()):
-            # skip if we already have user charges
-            if not (
-                off_mol.partial_charges is not None and np.any(off_mol.partial_charges)
-            ):
-                # due to issues with partial charge generation in ambertools
-                # we default to using the input conformer for charge generation
-                off_mol.assign_partial_charges(
-                    "am1bcc", use_conformers=off_mol.conformers
-                )
+        # Assign partial charges to all small mols
+        all_openff_mols = list(
+            chain(all_alchemical_mols.values(), common_small_mols.values())
+        )
+        self._assign_openff_partial_charges(
```

Comment:
I generally suggest adding a test like this to Protocol to ensure that user changes (which are the recommended way in which charges are going to be passed to this Protocol) work: https://github.com/OpenFreeEnergy/openfe/blob/main/openfe/tests/protocols/test_openmm_equil_rfe_protocols.py#L708

Reply:
This is a good idea, I'll be working on that asap. I added some tests to check that the actual values for the methods and backends are being passed and used, but a test on the actual assigned charges should work better.

Reply:
From meeting we want this to be more in terms of ensuring user charges are not changed anywhere in the protocol.

Reply:
Should be getting tested now in https://github.com/choderalab/feflow/blob/aec9428d1a943d156cedf3fe299f23545575058a/feflow/tests/test_nonequilibrium_cycling.py#L462